OSDOCS-13591: Documented 4.17.20 z-stream release notes (#89878)
openshift/openshift-docs: release_notes/ocp-4-17-release-notes.adoc

* Previously, the *Cluster Settings* page would not properly render during a cluster update if the `ClusterVersion` did not receive a `Completed` update.  With this release, the *Cluster Setting* page properly renders even if the `ClusterVersion` has not received a `Completed` update. (link:https://issues.redhat.com/browse/OCPBUGS-51292[*OCPBUGS-51292*])

* Previously, the alerts links on the *Alert rules* page of the *Developer* perspective included external labels to invalid links. This happened because the URL for the *Alerts* page did not expect external labels. With this release, the external labels are no longer added to the alerts URL on the *Alert rules* page so the alerts links are accurate. (link:https://issues.redhat.com/browse/OCPBUGS-51126[*OCPBUGS-51126*])

> * Previously, users could not create `NodePools` resources with ARM64 architecture on non-AWS or {azure-short} platforms. This bug resulted in validation errors that prevented the addition of bare-metal compute nodes and caused CEL validation blocks when creating a `NodePool` resource. The fix modifies the `NodePool` spec validation rules to allow ARM64 architecture on non-AWS or {azure-short} or Agent platforms by setting `None` for the `self.platform.type` section, so users can now create `NodePools` with ARM64 architecture specifications on non-AWS or {azure-short} bare-metal platforms, expanding functionality for end users. (link:https://issues.redhat.com/browse/OCPBUGS-46440[*OCPBUGS-46440*])

Reviewer: ```suggestion
* Previously, you could not create `NodePools` resources with ARM64 architecture on non-AWS or {azure-short} platforms. This bug resulted in validation errors that prevented the addition of bare-metal compute nodes and caused CEL validation blocks when creating a `NodePool` resource. The fix modifies the `NodePool` spec validation rules to allow ARM64 architecture on non-AWS, {azure-short}, or Agent platforms by setting `None` for the `self.platform.type` section. You can now create `NodePools` with ARM64 architecture specifications on non-AWS or {azure-short} bare-metal platforms. (link:https://issues.redhat.com/browse/OCPBUGS-46440[*OCPBUGS-46440*])
```

- **Note:** Agent platform and bare-metal platform are the same. I am wondering if it's _"Azure bare-metal platform"_ **OR** _"Azure or bare-metal (i.e Agent) platform"_
- IMO "expanding functionality for end users." is intended and can be removed.
- Do we need to expand CEL (validation blocks) here?
- [Reboot:](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#r)
> Reboot is widely recognized among technical users, but where usage permits, use restart.

> * Previously, for a `kubevirt-csi` pod that ran on a node in a guest cluster, the persistent volume claim (PVC) from the guest cluster was removed from the VM after the VM was rebooted. However, the `VolumeAttachment` resource was not removed and this caused issues for the cluster as it expected the PVC to be attached to the VM. With this release, after a VM is rebooted, the `VolumeAttachment` resource is removed so that the cluster issues no longer occur. (link:https://issues.redhat.com/browse/OCPBUGS-44623[*OCPBUGS-44623*])

Reviewer: ```suggestion
* Previously, for a `kubevirt-csi` pod that ran on a node in a guest cluster, the persistent volume claim (PVC) from the hosted cluster was removed from the virtual machine (VM) after the VM was restarted. However, the `VolumeAttachment` resource was not removed and this caused issues for the cluster as it expected the PVC to be attached to the VM. With this release, after a VM is rebooted, the `VolumeAttachment` resource is removed so that the cluster issues no longer occur. (link:https://issues.redhat.com/browse/OCPBUGS-44623[*OCPBUGS-44623*])
```

We use the term _**hosted cluster**_ rather than _**guest cluster**_ . Both terms mean the same.

> * Previously, in bare-metal configurations where the provisioning network is disabled but the `bootstrapProvisioningIP` field is set, the bare-metal provisioning components might fail to start. These failures occur when the provisioning process reconfigures the external network interface on the bootstrap VM during the process of pulling container images. With this release, dependencies were added to ensure that interface reconfiguration only occurs when the network is idle, preventing conflicts with other processes. As a result, the bare-metal provisioning components now start reliably, even when the `bootstrapProvisioningIP` field is set and the provisioning network is disabled. (link:https://issues.redhat.com/browse/OCPBUGS-43528[*OCPBUGS-43528*])

Reviewer: ```suggestion
* Previously, in bare-metal configurations where the provisioning network was disabled but the `bootstrapProvisioningIP` field was set, the bare-metal provisioning components would fail to start. These failures occur when the provisioning process reconfigures the external network interface on the bootstrap VM during the process of pulling container images. With this release, dependencies are added to ensure that interface reconfiguration only occurs when the network is idle, preventing conflicts with other processes. As a result, the bare-metal provisioning components now start reliably, even when the `bootstrapProvisioningIP` field is set and the provisioning network is disabled. (link:https://issues.redhat.com/browse/OCPBUGS-43528[*OCPBUGS-43528*])
```

Wondering if the past tense would fit here as suggested in the change.
